DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -77,13 +83,28 @@ function Set-TargetResource
         $AutoUpgrade = $false
     )
     Write-Verbose -Message 'Start Set-TargetResource'
+    $isVersionPresent = $PSBoundParameters.ContainsKey('Version')
+    $isMinimumVersionPresent = $PSBoundParameters.ContainsKey('MinimumVersion')
+    if ($isVersionPresent -and $isMinimumVersionPresent ) {
+        throw "Cannot specify 'Version' and 'MinimumVersion' in the same configuration"
+    }

```

Comment:
You should use a Parameter Set here rather than testing for it in the code - let PowerShell do the work for you.

Reply:
So I might be wrong here so if you know a fix then id love to know. But my understanding is that DSC doesn't support Parameter Sets, it will simply call Set-TargetResource with all the parameters a user has specified in the Configuration. So for this particular example if I were to remove this check the resource would prefer the Version parameter over the MinimumVersion parameter but it would be silent and may lead to confusion (at least it has already with our internal version of this resource)

Happy to discuss other options though.

Reply:
I genuinely didn't know that DSC doesn't support Parameter Sets. Never used them in DSC so it's not something I've come across as 'not working'. But yeah, thinking about it would change the parameters that are / can be passed to the `*-TargetResource` functions so they might all have different ones which isn't going to work.